### PR TITLE
Remove UpdateSettingsTestHelper class

### PR DESCRIPTION
**Pull request by dakrone**
By making the `settings()` method public on `UpdateSettingsRequest` (I think it
should have been in the first place) we can get rid of this class entirely. Mock
response objects are now constructed by parsing JSON without making the
constructor public.

Relates to #29823

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexlifecycle/SetSingleNodeAllocateStepTests.java [hunk 1]
```diff
@@ -340,17 +360,24 @@ private void assertNodeSelected(IndexMetaData indexMetaData, Index index, Set<St
 
         Mockito.when(client.admin()).thenReturn(adminClient);
         Mockito.when(adminClient.indices()).thenReturn(indicesClient);
+
+        final UpdateSettingsResponse response;
+        try (XContentParser parser = XContentType.JSON.xContent()
+                .createParser(NamedXContentRegistry.EMPTY, THROW_UNSUPPORTED_OPERATION, "{\"acknowledged\": true}")) {
+            response = UpdateSettingsResponse.fromXContent(parser);
```
Comment (talevy) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexlifecycle/SetSingleNodeAllocateStepTests.java]: 😄

Comment (talevy) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexlifecycle/SetSingleNodeAllocateStepTests.java]: I ended up using the `UpdateSettingsAction.INSTANCE.newResponse()` then `response.readFrom(StreamInput.wrap(new byte[] { ... }))` route for this workaround. But I like this too! more self-describing

**Review by talevy — APPROVED** [review]

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexlifecycle/SetSingleNodeAllocateStepTests.java (hunk 1 above)
Comment (colings86) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexlifecycle/SetSingleNodeAllocateStepTests.java]: Can we not make the Response constructor public here? Having the object parsed from a JSON string seems fragile and ugly to me.

**Review by colings86 — CHANGES_REQUESTED** [review]
I'm not sure about having the response parsed from a JSON string, it feels ugly to me and IMO more ugly than having the UpdateSettingsTestHelper class. I'm wondering if we can't just make the Response constructor public?

**Review by colings86 — APPROVED** [review]
LGTM

**Pull request merged**